iiod-client.c

```diff
@@ -280,8 +284,12 @@ struct iiod_client * iiod_client_new(const struct iio_context_params *params,
 
 void iiod_client_destroy(struct iiod_client *client)
 {
-	if (client->responder)
+	if (client->responder) {
+		if (client->ops->cancel)
+			client->ops->cancel(client->desc);
```

Comment:
nit: maybe we could introduce an helper to avoid the nested checks... Not that terrible though

Reply:
As long as it doesn't require gymnastics to fit under the 80-characters rule, and the pattern is not duplicated elsewhere, I think it can stay inline.